### ISSUE TITLE
ConvNeXt Architecture

### ISSUE THREAD

**Issue by LukeWood** [keras-team/keras-cv#68]
Sayak is working on this

**Comment by sayakpaul** [comment]
I have implemented it: https://github.com/sayakpaul/ConvNeXt-TF/. Should be available here soon: https://tfhub.dev/sayakpaul/collections/convnext/1.

**Comment by bhack** [comment]
I suppose that that one had an inital approach of converting the Pytorch reference impl weights.
It would be nice to see how we will handle this with the points we have at:
https://github.com/keras-team/keras-cv/issues/71

**Comment by sayakpaul** [comment]
Yeah, absolutely. If you serialize the weights of the converted models you will have something similar to what's expected here.

**Comment by bhack** [comment]
More ore less. I think that some of difference in this repo could be:

- As a library we could care to have a reusable components API of the network components. Probably more generic that just to build the original Networks and its variants that we could find in the same paper.

- The training scirpt, dataset interfaces API and markdown to reproduce the weights from scratch

- Extra (IMHO) finte tuning

- Extra++ Having a community CI infra/github self-hosted actions on GKE to launch user contributed training/fine-tuning jobs approved by maintainers revieiws.

**Comment by sayakpaul** [comment]
I am not absolutely sure about any of these points. I am also not sure how individual contributors could run the models from scratch without any involved support from the maintainers. 

For implementing architectures, there could be specific bits while generic bits could still benefit from what the library already offers. If you see what I have implemented, you'd probably notice there's not too many specific bits there. 

Also, it's helpful to have examples aiding your points. For example, if you could provide an example on what you meant in your first point in the context of this issue thread, that would be super helpful.

**Comment by bhack** [comment]
> I am not absolutely sure about any of these points. I am also not sure how individual contributors could run the models from scratch without any involved support from the maintainers.

We don't have this infra right now inpalce. So i we cannot contribute a Github action orchestrating the training job for reproducibility. 
It was just my perspective feature request (this why was tagget as extra++).

> For implementing architectures, there could be specific bits while generic bits could still benefit from what the library already offers. If you see what I have implemented, you'd probably notice there's not too many specific bits there. Also, it's helpful to have examples aiding your points. For example, if you could provide an example on what you meant in your first point in the context of this issue thread, that would be super helpful.

Yes other then thinking about obivious reusable components related to a specific network like new layers, optimizers, loss and metrics the It could introduce and that could be exposed here as API there was also:
https://github.com/keras-team/keras-cv/issues/59

So mainly It is just thinking as a library with the network as an e2e integration example of the new (if needed) introduced components API

**Comment by LukeWood** [comment]
Hey @sayakpaul just a heads up, we are planning to hold off on incorporating models for a little bit longer.  

@qlzh727 has some great ideas on changing the structure for Keras applications a bit, and we'd like to iron those out before adding any models.

**Comment by sayakpaul** [comment]
Thanks for letting me know. Does this also mean `keras.applications` will be held off from accepting new models for now?

**Comment by LukeWood** [comment]
> Thanks for letting me know. Does this also mean `keras.applications` will be held off from accepting new models for now?

I'd guess so, this should only last a month or so before we have the new sample model ready though.

**Comment by Tony363** [comment]
@LukeWood Is this issue open for contributions as well?! Next to SWIN transformers, ConvNext boasts even higher performance and stats utilizing similar robustness of training datasets!

**Comment by LukeWood** [comment]
It is not open for contributions as of now.  I believe Sayak will be contributing these to Keras core.  Is that correct @sayakpaul?  As of now KerasCV won't be accepting models until some details are ironed out on our end.

**Comment by sayakpaul** [comment]
I am working on it, yes.

**Comment by LukeWood** [comment]
@sayakpaul we can migrate this to KerasCV when you are ready.

**Comment by sayakpaul** [comment]
Yes sure. I will start working on it very soon.

**Comment by LukeWood** [comment]
Sayak fixed this!
